lib/beaker-pe/install/pe_utils.rb

```diff
@@ -450,7 +450,15 @@ def do_install hosts, opts = {}
               else
                 prepare_host_installer_options(host)
                 register_feature_flags!(opts)
-                generate_installer_conf_file_for(host, hosts, opts)
+
+                if opts[:upgrade] && use_meep?(host['previous_pe_ver'])
```

Comment:
So our upgrade cases are:

* from 2016.2.0+ (post-meep)
  * Beaker runs the installer such that we make use of recovery
* from < 2016.2.0 (pre-meep)
  * Beaker creates a fresh pe.conf using beaker-answers, as if we were doing an install